src/operator/nn/upsampling-inl.h

```diff
@@ -84,6 +86,23 @@ struct UpSamplingParam : public dmlc::Parameter<UpSamplingParam> {
   }
 };  // struct UpSamplingParam
 
+inline int* scaleComp(const UpSamplingParam &param) {
+  int* scaleArr = new int[2];
+  scaleArr[0] = 1;
+  scaleArr[1] = 1;
+  if (param.scale.ndim() == 1) {
+    scaleArr[0] = param.scale[0];
+    scaleArr[1] = param.scale[0];
+  } else if (param.scale.ndim() == 2) {
+    scaleArr[0] = param.scale[0];
+    scaleArr[1] = param.scale[1];
+  } else if (param.scale.ndim() == 4) {
+    scaleArr[0] = param.scale[2];
```

Comment:
Please elaborate this use case. How the control flow of the program reaches here ? What use case it handles. If its not intuitive please add comments to the code.

Reply:
I added a comment--I kept most of the previous implementation before this tuple scales were supported, and I believe this section is used to figure out the height/width scaling values when the multi-input mode is not activated.